packages/jest-jasmine2/src/index.js

```diff
@@ -69,7 +69,49 @@ function jasmine2(config, environment, moduleLoader, testPath) {
       moduleLoader.requireModule(null, config.setupTestFrameworkScriptFile);
     }
   });
+
+  const hasIterator = object => !!(object !== null && object[Symbol.iterator]);
+  const jestCustomEqualityTesters = [];
+  const iterableEquality = (a, b) => {
+    if (
+      typeof a !== 'object' ||
+      typeof b !== 'object' ||
+      Array.isArray(a) ||
+      Array.isArray(b) ||
+      ![a, b].every(hasIterator)
+    ) {
+      return undefined;
+    }
+    if (a.constructor !== b.constructor) {
+      return false;
```

Comment:
this check is only called for typeof === 'object', excluding arrays and that have iterators, so it should be fine like this, I'll check after importing and in case I can return `undefined` here.